File: server/ChatServer/ConversationKeys.cs
using System;

namespace ChatServer;

/// <summary>
/// 会话 ID 生成规则（客户端可本地推算，用于去重/历史拼接）。
/// 私聊: p_{guidA}_{guidB}（两 ID 按字符串升序）
/// 群聊: g_{groupId}
/// </summary>
public static class ConversationKeys
{
    public static string Private(Guid a, Guid b)
    {
        var (x, y) = a.CompareTo(b) < 0 ? (a, b) : (b, a);
        return $"p_{x}_{y}";
    }

    public static string Group(Guid groupId) => $"g_{groupId}";
}

File: client/flutter_chat/lib/utils/conversation_keys.dart
/// Conversation ID generation — mirrors server/ChatServer/ConversationKeys.cs.
/// Clients can compute these locally to match incoming SignalR messages.
///
/// Private: p_{guidA}_{guidB}  (two ids sorted ascending, string order)
/// Group:   g_{groupId}
String privateConversationId(String a, String b) {
  final sorted = [a, b]..sort();
  return 'p_${sorted[0]}_${sorted[1]}';
}

String groupConversationId(String groupId) => 'g_$groupId';
